Libraries/Opc.Ua.Security.Certificates/Common/Oids.cs

```diff
@@ -0,0 +1,157 @@
+
+        // Authority Inforamtion access
+        public const string AuthorityInfoAccess = "1.3.6.1.5.5.7.1.1";
+        public const string OnlineCertificateStatusProtocol = "1.3.6.1.5.5.7.48.1";
+        public const string CertificateAuthorityIssuers = "1.3.6.1.5.5.7.48.2";
+
+        public const string CRLDistributionPoint = "2.5.29.31";
+
+        /// <summary>
+        /// Get the RSA oid for a hash algorithm signature.
+        /// </summary>
+        /// <param name="hashAlgorithm">The hash algorithm name.</param>
+        public static string GetRSAOid(HashAlgorithmName hashAlgorithm)
+        {
+            if (hashAlgorithm == HashAlgorithmName.MD5)
+            {
+                return Oids.RsaPkcs1Md5;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA1)
+            {
+                return Oids.RsaPkcs1Sha1;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA256)
+            {
+                return Oids.RsaPkcs1Sha256;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA384)
+            {
+                return Oids.RsaPkcs1Sha384;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA512)
+            {
+                return Oids.RsaPkcs1Sha512;
+            }
+            else
+            {
+                throw new NotSupportedException($"Signing RSA with hash {hashAlgorithm.Name} is not supported. ");
+            }
+        }
+
+        /// <summary>
+        /// Get the ECDsa oid for a hash algorithm signature.
+        /// </summary>
+        /// <param name="hashAlgorithm">The hash algorithm name.</param>
+        public static string GetECDsaOid(HashAlgorithmName hashAlgorithm)
+        {
+            if (hashAlgorithm == HashAlgorithmName.SHA1)
+            {
+                return Oids.ECDsaWithSha1;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA256)
+            {
+                return Oids.ECDsaWithSha256;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA384)
+            {
+                return Oids.ECDsaWithSha384;
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA512)
+            {
+                return Oids.ECDsaWithSha512;
+            }
+            else
+            {
+                throw new NotSupportedException($"Signing ECDsa with hash {hashAlgorithm.Name} is not supported. ");
+            }
+        }
+
+        /// <summary>
+        /// Get the hash algorithm used to sign a certificate.
+        /// </summary>
+        /// <param name="oid">The signature algorithm oid.</param>
+        public static HashAlgorithmName GetHashAlgorithmName(string oid)
+        {
+            switch (oid)
+            {
+                case Oids.RsaPkcs1Md5: return HashAlgorithmName.MD5;
+                case Oids.ECDsaWithSha1:
+                case Oids.RsaPkcs1Sha1: return HashAlgorithmName.SHA1;
```

Comment:
Random comment about style.  If this was all one case label to one value, I'd understand pulling up the return, but for these multi-ones it feels wrong and makes it harder (for me) to read.

```suggestion
                case Oids.ECDsaWithSha1:
                case Oids.RsaPkcs1Sha1:
                    return HashAlgorithmName.SHA1;
```

Looks much clearer to me.  And, of course, once one thing isn't using the lifted style none should.

Reply:
done